src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java

```diff
@@ -151,33 +151,34 @@ public ActivityServiceImpl(ActivityRepository activityRepository, RecordsService
     public Page<ProviderRecordDTO> getPartnerActivitiesForCurrentUser(Pageable pageable) {
         UserProfile userProfile = userService.getCurrentUserProfile();
         Set<UserGroup> userGroups = userProfile.getUserGroups();
-        Page<Organization> organizations;
         if (userGroups == null || userGroups.isEmpty()) {
-            organizations = providerRecordsRepository
+            return providerRecordsRepository
```

Comment:
you should use the filterProviderRecords here (to remove the excluded fields and locations)